docs/howto/partitions.rst

```diff
@@ -0,0 +1,167 @@
+******************************************
+Add partition support to an application
+******************************************
+
+Partitions basics
+=================
+
+Applications implementing ``craft-application`` and its dependent suite of
+libraries can optionally make use of `disk partitions
+<https://en.wikipedia.org/wiki/Disk_partitioning>`_.  The supported list of
```

Comment:
Craft partitioning is not necessarily used to map disk partitions. In Snapcraft they're used to implement components as separate output artifacts, and other applications may use it for other purposes (e.g. Rockcraft could use partitions to create additional image layers).

Reply:
Indeed, we should mention disk partitions as one example use case but make it clear that it's more general than that